admin/test/utils.go

```diff
@@ -0,0 +1,113 @@
+/*
+ *
+ * Copyright 2021 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+// Package test contains test only functions for package admin. It's used by
+// admin/admin_test.go and admin/test/admin_test.go in this directory.
+package test
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"time"
+
+	v3statuspb "github.com/envoyproxy/go-control-plane/envoy/service/status/v3"
+	"github.com/google/uuid"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/admin"
+	channelzpb "google.golang.org/grpc/channelz/grpc_channelz_v1"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/internal/xds"
+	"google.golang.org/grpc/status"
+)
+
+const (
+	defaultTestTimeout = 10 * time.Second
+)
+
+// RunAndCode contains the function to make an RPC and the expected status code
+// (can be OK).
+type RunAndCode struct {
```

Comment:
As a parameter to `RunRegisterTests`, this might be better as:

```go
type RegisterTestCases struct {
  ChannelzCodeWant codes.Code
  CSDSCodeWant codes.Code
}
```

(or something)

Reply:
Done